contributors/design-proposals/node-allocatable.md

```diff
@@ -89,61 +73,253 @@ The flag will be specified as a serialized `ResourceList`, with resources define
+
+**Status**: Implemented v1.2
+
+In this phase, Kubelet will support specifying `KubeReserved` & `SystemReserved` resource reservations via kubelet flags.
+The defaults for these flags will be `""`, meaning zero cpu or memory reservations.
+Kubelet will compute `Allocatable` and update `Node.Status` to include it.
+The scheduler will use `Allocatable` instead of `Capacity` if it is available.
+
+### Phase 2 - Enforce Allocatable on Pods
+
+**Status**: Targeted for v1.6
+
+In this phase, Kubelet will automatically create a top level cgroup to enforce Node Allocatable across all user pods.
+The creation of this cgroup is controlled by `--cgroups-per-qos` flag.
+
+Kubelet will support specifying the top level cgroups for `KubeReserved` and `SystemReserved` and support *optionally* placing resource restrictions on these top level cgroups.
+
+Users are expected to specify `KubeReserved` and `SystemReserved` based on their deployment requirements.
+
+Resource requirements for Kubelet and the runtime is typically proportional to the number of pods running on a node.
+Once a user identified the maximum pod density for each of their nodes, they will be able to compute `KubeReserved` using [this performance dashboard](http://node-perf-dash.k8s.io/#/builds).
+[This blog post](http://blog.kubernetes.io/2016/11/visualize-kubelet-performance-with-node-dashboard.html) explains how the dashboard has to be interpreted.
+Note that this dashboard provides usage metrics for docker runtime only as of now.
+
+New flags introduced in this phase are as follows:
 
-## Issues
+1. `--enforce-node-allocatable=[pods][,][kube-reserved][,][system-reserved]`
+
+	* This flag will default to `pods` in v1.6.
+	* This flag will be a `no-op` unless `--kube-reserved` and/or `--system-reserved` has been specified.
+	* If `--cgroups-per-qos=false`, then this flag has to be set to `""`. Otherwise its an error and kubelet will fail.
+	* It is recommended to drain and restart nodes prior to upgrading to v1.6. This is necessary for `--cgroups-per-qos` feature anyways which is expected to be turned on by default in `v1.6`.
+	* Users intending to turn off this feature can set this flag to `""`.
+	* Specifying `kube-reserved` value in this flag is invalid if `--kube-reserved-cgroup` flag is not specified.
+	* Specifying `system-reserved` value in this flag is invalid if `--system-reserved-cgroup` flag is not specified.
+	* By including `kube-reserved` or `system-reserved` in this flag's value, and by specifying the following two flags, Kubelet will attempt to enforce the reservations specified via `--kube-reserved` & `system-reserved` respectively.
+
+2. `--kube-reserved-cgroup=<absolute path to a cgroup>`
+   * This flag helps kubelet identify the control group managing all kube components like Kubelet & container runtime that fall under the `KubeReserved` reservation.
+
+3. `--system-reserved-cgroup=<absolute path to a cgroup>`
```

Comment:
Defaulting SGTM. It will help with metrics collection. I'd tackle that in v1.7 once all the distro owners digest this proposal.